proto/osmosis/lockup/lock.proto

```diff
@@ -54,16 +54,16 @@ message PeriodLock {
 enum LockQueryType {
   option (gogoproto.goproto_enum_prefix) = false;
 
-  ByDuration = 0;
-  ByTime = 1;
+  ByTime = 0;
+  ByDuration = 1;
```

Comment:
why was this changed again?

Reply:
Even if we pass `lockuptypes.QueryCondition{}` to the function, it will still get the default value of `distrTo.LockQueryType` which is the first value of `LockQueryType` (In this case `ByDuration` => thus causing an error in the loop. ) 
So I just change the index to bypass.

Reply:
for CL the we are not distributing via locktypes so, what i originally thought was that we could skip distrTo check  below; 
```
	if distrTo.LockQueryType == lockuptypes.ByDuration {
		durationOk := false
		for _, duration := range durations {
			if duration == distrTo.Duration {
				durationOk = true
				break
			}
		}
		if !durationOk {
			return 0, fmt.Errorf("invalid duration: %d", distrTo.Duration)
		}
	}

	// check if denom this gauge pays out to exists on-chain
	if distrTo.Denom != "" && !k.bk.HasSupply(ctx, distrTo.Denom) && !strings.Contains(distrTo.Denom, "osmovaloper") {
		return 0, fmt.Errorf("denom does not exist: %s", distrTo.Denom)
	}

```

by passing empty queryCondition. lmk if that makes sense

Reply:
Yeah I understand ur idea. But the problem I face is even calling `CreateGauge` with empty queryCondition. It is still assigned a value in the function.
Thinking about adding new value to `LockQueryType` enum to bypass both ByDuration & ByTime. Lmk what u think about that

Reply:
i see what about changing the enumType to something like;
```
enum LockQueryType {
  option (gogoproto.goproto_enum_prefix) = false;

  UNSPECIFIED = 0;
  ByTime = 0;
  ByDuration = 1;
}

```
if this works, hopefully we donot have to make lots of core logic changes and it works with existing queryCondition

Reply:
Yeah we r same page 🙌